Sources/XCHammer/Generator.swift

```diff
@@ -908,10 +908,9 @@ enum Generator {
                  fatalError("Can't write genStatus")
             }
         }
-        let result: Result<(), GenerateError> = .success()
-        return results.reduce(into: result, {
+        return results.reduce(.success()) {
              result, element in
-             return result &&& element
-        })
+             return result.flatMap { _ in element }
```

Comment:
Ok interesting - so `Result` has it's own `flatMap` that will effectively achieve the same result - this LGTM 👍

Reply:
Yeah, and `Result`'s `flatMap` propagates a failure. The result of this `reduce` + `flatMap` is a result that is success if all results were successes, or the first failure found. Based on the caller of this function, this seems like the best implementation.